internal/webapi/webapi.go

```diff
@@ -122,48 +130,46 @@ func Start(ctx context.Context, requestShutdown func(), shutdownWg *sync.WaitGro
 		cache:       cache,
 		signPrivKey: signPrivKey,
 		signPubKey:  signPubKey,
+		listener:    listener,
 	}
 
-	// Create TCP listener.
-	var listenConfig net.ListenConfig
-	listener, err := listenConfig.Listen(ctx, "tcp", cfg.Listen)
-	if err != nil {
-		return err
-	}
-	log.Infof("Listening on %s", cfg.Listen)
-
-	srv := http.Server{
+	w.server = &http.Server{
 		Handler:      w.router(cookieSecret, dcrd, wallets),
 		ReadTimeout:  5 * time.Second,  // slow requests should not hold connections opened
 		WriteTimeout: 60 * time.Second, // hung responses must die
 	}
 
+	return w, nil
+}
+
+func (w *WebAPI) Run(ctx context.Context, requestShutdown func(), shutdownWg *sync.WaitGroup) {
+
 	// Add the graceful shutdown to the waitgroup.
 	shutdownWg.Add(1)
 	go func() {
 		// Wait until shutdown is signaled before shutting down.
 		<-ctx.Done()
 
-		log.Debug("Stopping webserver...")
-		// Give the webserver 5 seconds to finish what it is doing.
+		w.log.Debug("Stopping webserver...")
+		// Give the webserver 10 seconds to finish what it is doing.
```

Comment:
I'm fairly certain this is not necessary.  See the explanation in the related [dcrpool PR](https://github.com/decred/dcrpool/pull/371#discussion_r1327507407).

Reply:
I've been [playing around](https://go.dev/play/p/-F2Mrj8ft2Z) with this and you're totally right. Updated this in a new commit.